### PR TITLE
[GEOS-8915] WIP first support for ssl in postgresql

### PR DESCRIPTION
**Pull request by RemiDesgrange**

### CODE REVIEW

**Comment by tbarsballe** [comment]
Closing and re-opening in an attempt to fix the Travis CI failure.

**Comment by tbarsballe** [comment]
Remaining travis failure appears to be legitimate - compile failure in gt-jdbc-postgis
This needs to be fixed before this PR can be merged.

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java [hunk 1]
```diff
@@ -16,9 +16,12 @@
  */
 package org.geotools.data.postgis;
 
+import static com.sun.tools.doclint.Entity.and;
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java]: These imports should not be there.

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java [hunk 2]
```diff
@@ -336,18 +407,22 @@ private Connection getConnection(String user, String password, String url) throw
     public void dropDatabase(Map<String, Object> params) throws IOException {
         JDBCDataStore closer = new JDBCDataStore();
         // get the connection params
-        String host = (String) HOST.lookUp(params);
-        int port = (Integer) PORT.lookUp(params);
         String db = (String) DATABASE.lookUp(params);
         String user = (String) USER.lookUp(params);
         String password = (String) PASSWD.lookUp(params);
+        // TODO make a copy of the map (hash map ?)
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java]: Please don't modify the values provided by user, the database.key should not be touched.

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java [hunk 3]
```diff
@@ -234,7 +295,16 @@ protected String getJDBCUrl(Map params) throws IOException {
         String host = (String) HOST.lookUp(params);
         String db = (String) DATABASE.lookUp(params);
         int port = (Integer) PORT.lookUp(params);
-        return "jdbc:postgresql" + "://" + host + ":" + port + "/" + db;
+        SslMode sslMode = (SslMode) SSL_MODE.lookUp(params);
+        Path sslCa =  (Path)SSL_ROOTCERT.lookUp(params);
+        Path sslKey = (Path) SSL_KEY.lookUp(params);
+        Path sslCert = (Path) SSL_CERT.lookUp(params);
+        String jdbcUrl = String.format("jdbc:postgresql://%s:%s/%s", host, port, db);
+        jdbcUrl = String.format("%s?sslmode=%s", jdbcUrl, sslMode.value);
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java]: If SSL is not active I'd avoid adding any SSL related bit

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java [hunk 4]
```diff
@@ -32,11 +35,17 @@
 import org.geotools.jdbc.JDBCDataStoreFactory;
 import org.geotools.jdbc.SQLDialect;
 import org.geotools.util.KVP;
+import org.postgresql.jdbc.SslMode;
 
-/** @source $URL$ */
+
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java]: Avoid any manual reformat, let the automatic formatting do the work. PR with manual reformats will be rejected.

**Comment by RemiDesgrange** [comment]
@tbarsballe I need to create test first before anything. And fix the typo/bug that @aaime pointed

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java (hunk 2 above)
Comment (RemiDesgrange) [reply modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java]: The idea here, is to copy the user input, in order to change the db from something to template1. I would never change the initial value of the object, just making a local copy of it, in order to pass this specific case.

**Comment by tbarsballe** [comment]
This now has merge conflicts, please fix the conflicts (along with all the other requested changes) so this can be merged.

**Comment by aaime** [comment]
@RemiDesgrange what should we do about this PR? It needs further work, and a CLA.

**Comment by RemiDesgrange** [comment]
Sorry. I don't have time to develop further ATM. I can rebase If you want, or you can just close the PR. I'm now part of camptocamp and will see if they are willing to sign the CLA and let me work on this. But it will not be before summer (at least).

**Comment by tbarsballe** [comment]
This functionality has been added by https://github.com/geotools/geotools/pull/2484
Closing this one due lack of activity

**Pull request closed without merging**